git/client.go

```diff
@@ -595,6 +596,9 @@ func (c *Client) Clone(ctx context.Context, cloneURL string, args []string, mods
 	} else {
 		target = path.Base(strings.TrimSuffix(cloneURL, ".git"))
 	}
+	if slices.Contains(cloneArgs, "--bare") {
+		target += ".git"
```

Comment:
I don't know if this is quite right. Just writing a comment here to ensure this isn't merged before I have a moment to investigate.

Reply:
Yes, as I suspected this is broken in the case where you provide both `--bare` **and** a target directory i.e.:

```
➜  cli git:(support-bear) ./bin/gh repo clone hyperrealist/tiled tiled -- --bare
Cloning into bare repository 'tiled'...
remote: Enumerating objects: 16472, done.
remote: Counting objects: 100% (376/376), done.
remote: Compressing objects: 100% (113/113), done.
remote: Total 16472 (delta 283), reused 336 (delta 263), pack-reused 16096
Receiving objects: 100% (16472/16472), 4.30 MiB | 11.09 MiB/s, done.
Resolving deltas: 100% (12482/12482), done.
failed to run git: fatal: cannot change to 'tiled.git': No such file or directory
```

We should only be appending `.git` to the target directory if the user didn't specify one.